### DISCUSSION TITLE
Smart contract developer portal

### DISCUSSION THREAD

**Discussion by greenhat** [0xPolygonMiden/compiler#167] [Ideas]
### Motivation
I'm trying to put myself in the shoes of a developer who wants to start developing smart contracts on Miden. I'm thinking of a one-page entry point into the ecosystem (portal). We already have the https://docs.polygon.technology/miden/ (Polygon Knowledge Layer) so this page might be a part of it. This portal should have links to the following:

### Guides on smart contract development in Rust. 
Various guides and tutorials on how to develop smart contracts in Rust, explaining Miden SDK (stdlib, compiler intrinsics, tx kernel API) with tutorials and best practices. The beginner's level guides and tutorials can be referenced directly on the portal page, and more advanced ones hidden behind a link to avoid information overload.

### Miden SDK API Reference

Crate documentation for Miden SDK crate(s) published on docs.rs would be the "source of truth" since it has the least chance to become outdated. We should strive to have it as extensive as possible, with examples and explanations for every function and type. 

### Examples of smart contracts

We have a new [contract-examples](https://github.com/0xPolygonMiden/contract-examples) repo that I intend to fill with examples of smart contracts. Every example will live in its own folder and have a README with a brief description. They can be cloned, built, and used as a starting point for building user's contracts. To make sure that the examples are up-to-date and working, we should have CI/CD that builds and tests them on every commit using the latest stable version of the toolchain.

## Other implementation details
We should put the link to this portal in all the relevant repos/crates - contract-examples, SDK, rust-templates along with blog posts and any articles that we write about Miden. 

Please let me know if you have any suggestions or ideas on how to improve this portal.

**Comment by bobbinth** [comment]
Agreed, I think overall we need to refactor how our documentation is shown in https://docs.polygon.technology/miden. For example, I think the structure could be as follows:
```
Miden
├── Overview
├── Get started
│   ├── Core concepts
│   └── Tutorials
├── How to
│   ├── Write a smart contract
│   └── Deploy a smart contract
├── Smart contracts
│   ├── Overview
│   ├── Miden SDK
│   └── Examples
├── Miden node
│   ├── Overview
│   ├── Deployment
│   └── API
├── Miden client
│   ├── Overview
│   ├── CLI
│   └── Library
└── Architecture
    ├── Accounts
    ├── Notes
    ├── Assets
    ├── Transactions
    ├── State
    └── Kernels
```
In the above "How to" and "Smart contracts" sections would be specific to smart contract development (though, maybe we can merge them into a single section).

We should also probably think about how to integrate smart contract development into Miden playground and/or the block explorer.

cc @Dominik1999.
